Start the application before running the tests

toland opened the pull request:
This mimics the behavior of exunit and turns out to be really useful. If that behavior isn't what is wanted, it can be disabled by passing `--no-start`.

Commit:
Start the application before running the tests

This mimics the behavior of exunit and turns out to be really useful. If that behavior isn't what is wanted, it can be disabled by passing `--no-start`.

dantswain commented:
Hm, I like the idea of this but I think it might be better to have it disabled by default.  People who have existing eunit tests probably have code already that starts their apps or supervisors during setup.  That is, after all, one of the "fun" parts of Erlang that Elixir improved upon ;)

It might be best to have a command line option as well as a setting that can be specified in the mix project config (i.e., so you can set it at a project level instead of having to pass in the command line argument every time).

toland commented:
That is a good point. I think we would want a setting in the `mix.exs` file since you are likely to want that every time. It shouldn't be hard to implement.

toland commented:
I know how to do this cleanly, but I am afraid that the solution is involved enough that it will cause merge conflicts with the other open PRs. What I would like to do is see if we can't get a couple of the other open PRs merged, then I can update this branch from master and keep the PR easily mergeable.

dantswain commented:
Makes sense to me.  I'll try to be responsive.

dantswain commented:
Merged #5 should be able to merge #8 with some suggested refactoring.  Feel free to hit me on elixir-lang slack - @dantswain there.

Commit:
Merge branch 'master' into start_app

* master:
  Make it clearer what the coverage code is doing
  Replace List.keymember? with Keyword.has_key?
  Handle coverage the same was as exunit
  Add eunit_formatters for better output
  Make sure test failures cause the task to fail
  Run the tests all at once rather than individually

# Conflicts:
#	lib/mix/tasks/eunit.ex

Commit:
Tighten up the option parsing

Use strict parsing so that typos or unkown options will cause an error to be raised.

Commit:
Control compile and app start with options

Commit:
Allow setting eunit options in the mixfile

Supports the verbose, profile, cover, start, and color options.

toland commented:
The application start is now disabled by default and you can set options in the mixfile.

Commit:
Fix a bug when there are no options in the mixfile

dantswain commented:
I think this all looks good.  I hate to keep dragging you along @toland but could you update the README wrt the new options and defaults?

toland commented:
No worries. I will update the docs now. Shouldn't take a moment.

Commit:
Update documentation and README

toland commented:
Done.

Commit:
Fix section headers in moduledoc

dantswain commented:
:+1: Thanks!
